src/ucp/wireup/wireup_cm.c

```diff
@@ -78,15 +78,30 @@ static unsigned ucp_cm_client_try_next_cm_progress(void *arg)
 
 static int ucp_cm_client_try_fallback_cms(ucp_ep_h ep)
 {
-    ucp_worker_h worker         = ep->worker;
-    ucp_rsc_index_t cm_idx      = ucp_ep_ext_control(ep)->cm_idx;
-    ucp_rsc_index_t next_cm_idx = cm_idx + 1;
-    uct_worker_cb_id_t prog_id  = UCS_CALLBACKQ_ID_NULL;
+    ucp_worker_h worker          = ep->worker;
+    ucp_rsc_index_t cm_idx       = ucp_ep_ext_control(ep)->cm_idx;
+    ucp_rsc_index_t next_cm_idx  = cm_idx + 1;
+    uct_worker_cb_id_t prog_id   = UCS_CALLBACKQ_ID_NULL;
+    ucp_rsc_index_t num_cm_cmpts = ucp_worker_num_cm_cmpts(worker);
+    ucs_string_buffer_t cms_strb;
+    int i;
+
+    if (next_cm_idx >= num_cm_cmpts) {
+        ucs_string_buffer_init(&cms_strb);
+
+        for (i = 0; i < num_cm_cmpts; ++i) {
+            if (i > 0) {
+                ucs_string_buffer_appendf(&cms_strb, ",");
+            }
+            ucs_string_buffer_appendf(&cms_strb, "%s",
+                                      ucp_context_cm_name(worker->context, i));
+        }
+
+        ucs_error("client ep %p failed to connect using %s cms, set "
+                  "UCX_LOG_LEVEL=diag for more details",
```

Comment:
I do not think that lack of device in `tl_bitmap` is the only case when client may not connect to the server IP address,
on other hand, this is good point to convert all such messages to "diag" if any 👍